src/components/posts/PostsPage.js

```diff
@@ -1,71 +1,154 @@
-import React from 'react';
-import { useState, useEffect } from 'react';
-import PostForm from './PostForm';
-import client from '../../utils/client';
-import './style.css';
-import { Box, Stack } from '@mui/material';
-import Header from '../Header/Header';
-import dateTimetoRelativeTime from './helperfunctions';
-import { Link } from 'react-router-dom';
+import React from 'react'
+import { useState, useEffect } from 'react'
+import PostForm from './PostForm'
+import client from '../../utils/client'
+import './style.css'
+import { Box, Stack } from '@mui/material'
+import Header from '../Header/Header'
+import dateTimetoRelativeTime from './helperfunctions'
+import { Link } from 'react-router-dom'
 
 const PostsPage = ({ role }) => {
-  const [post, setPost] = useState({ content: '' });
-  const [postResponse, setPostResponse] = useState('');
-  const [posts, setPosts] = useState([]);
+  const [post, setPost] = useState({ content: '' })
+  const [postResponse, setPostResponse] = useState('')
+  const [posts, setPosts] = useState([])
+  const [comment, setComment] = useState('')
+  const [load, setLoad] = useState(true)
+  const [showAllArr, setShowAll] = useState([])
 
   useEffect(() => {
-    client.get('/posts').then((res) => setPosts(res.data.data.posts));
-  }, []);
-
+    client.get('/posts').then((res) => {
+      setPosts(res.data.data.posts)
+    })
+  }, [load])
   const createPost = async (event) => {
-    event.preventDefault();
+    event.preventDefault()
     client
       .post('/post', post)
       .then((res) => {
-        setPostResponse(res.data);
-        setPosts((posts) => [res.data.data.post, ...posts]);
+        setPostResponse(res.data)
+        setPosts((posts) => [res.data.data.post, ...posts])
+        setLoad((x) => !x)
```

Comment:
As below, change `x` to a more meaningful variable name.

Reply:
DONE!